### PR TITLE
Filter environment

### PR DESCRIPTION
**Pull request by gimenete**
Adds a filter environment to be used for deployment events for example.

### CODE REVIEW

**Comment by hfaulds** [comment]
Hello @gimenete, thanks for the contribution. If you have time could you add a test. I'm also curious about the exact use-case, what ENV var do you have in mind to check against (I'm just trying to work out if a more specific filter might make sense)?

**Comment by gimenete** [comment]
@hfaulds the use case I'm looking to solve is triggering a workflow when there's a deployment to github pages. In this case github creates a deployment with environment=gh-pages. But I can think of other use cases, like sending an SMS using twilio when there's a deployment to production, or things like that.

I'll add a test as soon as possible!

**Comment by gimenete** [comment]
@parkr I just added some tests! Can you review again? Thanks!

File: filter/test/environment.bats [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+@test "environment: matches a or b" {
+  run environment "production|staging"
+  echo $output
+  [ "$status" -eq 0 ]
+  [ "$output" = "" ]
+}
+
+@test "environment: does not match a or b" {
+  run environment "development|staging"
+  echo $output
```
Comment (parkr) [inline filter/test/environment.bats]: Do you need these echo statements?

Comment (gimenete) [reply filter/test/environment.bats]: I thought it as a convention. I used this as a guide https://github.com/actions/bin/blob/master/filter/test/actor.bats and there are echo calls there. I can remove mine and those too if you want.

Comment (gimenete) [reply filter/test/environment.bats]: I've removed my echo statements and others.

**Pull request merged**